src/api/spec/helpers/validation_helper_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe ValidationHelper do
+  describe '#valid_project_name?' do
+    it 'calls Project.valid_name?' do
+      expect(Project).to receive(:valid_name?).with("home:mschnitzer")
+      valid_project_name?("home:mschnitzer")
+    end
+  end
+
+  describe '#valid_project_name!' do
+    it 'raises an exception if a project name is invalid' do
+      expect { valid_project_name!("home mschnitzer") }.to raise_error(ValidationHelper::InvalidProjectNameError)
```

Comment:
In the end it doesn't make sense to test all situations where project name can be invalid. The RegEx would be to complex I guess. Besides that, it's not the functions task to test all scenarios of valid project names. For that we've https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/models/project_spec.rb#L100